src/dbnode/storage/index/results_new_map.go

```diff
@@ -32,17 +34,16 @@ const (
 
 func newResultsMap(idPool ident.Pool) *ResultsMap {
 	return _ResultsMapAlloc(_ResultsMapOptions{
-		hash: func(k ident.ID) ResultsMapHash {
-			return ResultsMapHash(xxhash.Sum64(k.Bytes()))
-		},
-		equals: func(x, y ident.ID) bool {
-			return x.Equal(y)
+		hash: func(k []byte) ResultsMapHash {
+			return ResultsMapHash(xxhash.Sum64(k))
 		},
-		copy: func(k ident.ID) ident.ID {
-			return idPool.Clone(k)
+		equals: bytes.Equal,
+		// TODO(nate); are these copy and finalize implementations right?
+		copy: func(k []byte) []byte {
```

Comment:
Explicitly calling out the `copy` and `finalize` implementations here for review feedback

Reply:
Is the ID `[]byte` key that gets passed in initially also from this `idPool`? In general this look right to me but if the incoming key isn't pooled then maybe we wouldn't want the copy to come from `idPool`.

Reply:
Ah, yeah, initial `[]byte` is not from a pool. In fact, the one call site that would invoke this method passes in the `NoCopy` option, so this method won't even be called.